var/spack/repos/builtin/packages/sopt/package.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Sopt(CMakePackage):
+    """SOPT is an open-source C++ package available under the license below. It performs
+    Sparse OPTimisation using state-of-the-art convex optimisation algorithms. It solves a
+    variety of sparse regularisation problems, including the Sparsity Averaging Reweighted
+    Analysis (SARA) algorithm.
+    """
+
+    homepage = "https://astro-informatics.github.io/sopt/"
+    url = "https://github.com/astro-informatics/sopt/archive/refs/tags/v4.2.0.tar.gz"
+    git = "https://github.com/astro-informatics/sopt"
+
+    maintainers("tkoskela", "mmcleod89", "20DM")
+    license("GPL-2.0")
+
+    version("4.2.0", sha256="25e579722f8e049d37c9155affa57ec2f38a2f8414c9cf430da2b7bafc86907b")
+
+    variant("tests", default=False, description="Build tests")
+    variant("examples", default=False, description="Build examples")
+    variant("benchmarks", default=False, description="Build benchmarks")
+    variant("openmp", default=False, description="Enable multithreading with OpenMP")
+    variant("mpi", default=False, description="Enable parallelisation with MPI")
+    variant("docs", default=False, description="Enable multithreading with OpenMP")
+    variant("coverage", default=False, description="Enable code coverage")
+    variant("onnxrt", default=False, description="Build with Tensorflow support using onnx")
+
+    depends_on("cmake@3")
+    depends_on("eigen@3.4")
+    depends_on("libtiff@4.5")
+    depends_on("mpi", when="+mpi")
+    depends_on("catch2@3.4", when="+tests")
+    depends_on("benchmark@1.8", when="+benchmarks")
+    depends_on("onnx@1.16", when="+onnxrt")
+    depends_on("doxygen@1.9", when="+docs")
```

Comment:
I have a problem building this specific version of `doxygen` with `gcc-13`:
```py
/home/bkaindl/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/spdlog-1.14.1-go7oyje5evdqx44xoiqvxbwbl4mijqy7/include/spdlog/fmt/fmt.h:24:14: fatal error: spdlog/fmt/bundled/core.h: No such file or directory
   24 |     #include <spdlog/fmt/bundled/core.h>
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

One can't build using gcc-13 with this version of doxygen, it seems.

- Therefore, see it as a problem to depend on this specific version of doxygen
  - Doxygen likely just misses a `conficts("%gcc@13:", when="@:1.9")` (or so, depends on specifics)

- Do you really need this specific version of doxygen (and specifically, only this version)?
  - This also applies to all the dependencies above
  - Usually, you should never depend on a single specific version, usually you depend on a range, that is usually open-ended: at least version x.x, or up to version x.x.

Reply:
It looks like we can build the documentation just fine with doxygen@1.12 (that doesn't give me any issues with gcc-13), so happy to remove that specific version. Is it better to have "at least version 1.12" in the spec, or leave it unspecified?

Reply:
I've added version ranges where it seemed reasonable to do so.